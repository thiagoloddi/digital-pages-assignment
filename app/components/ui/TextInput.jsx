import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

const style = {
  container: {
    '& > *': {
      display: 'block',
      width: '100%',
      textAlign: 'left'
    }
  },
  error: {
    marginTop: '5px',
    fontSize: '1rem',
    color: '#B00020'
  },
  input: {
    border: '1px solid black',
    padding: '5px'
  },
  inputError: {
    borderColor: '#B00020'
  },
}

const TextInput = ({ label, value, error, id, onChange, className, classes, type="text" }) => {
  return (
    <div className={` ${classes.container} ${className}`}>
      <label>{label}</label>
      <input className={`${classes.input} ${error ? classes.inputError : ''}`} id={id} value={value} onChange={onChange} type={type}/> 
      <div className={classes.error}>{error || ''}&nbsp;</div>
    </div>
  );
}

export default injectSheet(style)(TextInput);