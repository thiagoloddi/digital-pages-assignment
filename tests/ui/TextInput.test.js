import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from '../../app/components/ui/TextInput';

const func = () => {};

it('renders correctly without error', () => {
  const tree = renderer
    .create(<TextInput label="some label" value="some value" id="123" onChange={func} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly with error', () => {
  const tree = renderer
    .create(<TextInput label="some label" value="some value" error="some error" id="123" onChange={func} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});