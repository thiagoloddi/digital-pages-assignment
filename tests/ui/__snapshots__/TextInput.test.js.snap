// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with error 1`] = `
<div
  className=" TextInput-container-0-1-1 undefined"
>
  <label>
    some label
  </label>
  <input
    className="TextInput-input-0-1-3 TextInput-inputError-0-1-4"
    id="123"
    onChange={[Function]}
    type="text"
    value="some value"
  />
  <div
    className="TextInput-error-0-1-2"
  >
    some error
     
  </div>
</div>
`;

exports[`renders correctly without error 1`] = `
<div
  className=" TextInput-container-0-1-1 undefined"
>
  <label>
    some label
  </label>
  <input
    className="TextInput-input-0-1-3 "
    id="123"
    onChange={[Function]}
    type="text"
    value="some value"
  />
  <div
    className="TextInput-error-0-1-2"
  >
    
     
  </div>
</div>
`;
